test/init/projectAutoDetection.test.ts

```diff
@@ -118,9 +118,57 @@ describe('AutoDetectProject: tests', function () {
               'vscode': '',
             },
           };
-          expect(adp.isProject(packageJson, 'meteor')).to.be.false;
+          expect(adp.getInfo(packageJson, 'meteor')).to.be.null;
         });
 
+      it('does not detect a project when not project json is provided',
+        function () {
+          expect(adp.getInfo(null, '')).to.be.null;
+        });
+
+      it('gets the project info version correctly',
+        function () {
+          const packageJson = {
+            dependencies: {
+              '@angular/core': '0.0.0',
+            },
+          };
+          const projectInfo: models.IProjectInfo = adp.getInfo(packageJson, 'ng');
+          expect(projectInfo).to.not.be.null;
+          expect(projectInfo.name).to.equal('Angular');
+          expect(projectInfo.version).to.equal('0.0.0');
+        });
+
+      context('gets from search result', function () {
+        let sandbox: sinon.SinonSandbox;
+
+        beforeEach(() => {
+          sandbox = sinon.sandbox.create();
+        });
+
+        afterEach(() => {
+          sandbox.restore();
+        });
+
+        it('the project info if it matches',
+          function () {
+            sandbox.stub(fs, 'readFileSync').returns('{ "dependencies": { "@angular/core": "0.0.0" } }');
+            const results = [{ fsPath: '' }] as models.IVSCodeUri[];
+            const projectInfo: models.IProjectInfo = adp.getProjectInfo(results, 'ng');
```

Comment:
Why are we using different keys (ng & Angular) for `Angular`? I think we should unify these identifiers.

Reply:
The key is `ng` and `Angular` is an optional name for the project info. Currently, project info name is not used anywhere but, you know me, I thought it was nice to have.